### PR TITLE
docs: Add "Switching from 36.4 to 52.0" section [v2]

### PR DESCRIPTION
**Pull request by ldoktor**
This section should help people running 36.x to move towards using 52.x
Avocado version.

v1: https://github.com/avocado-framework/avocado/pull/2080

Changes

```yaml
v2: Move the doc to the "release_notes/lts/52_0.rst"
v2: Some really minor refinements to suite the new structure
```

### CODE REVIEW

**Comment by ldoktor** [comment]
[Documentation on how to switch from 36lts to lts-next](https://trello.com/c/ymL7SXYF/1001-documentation-on-how-to-switch-from-36lts-to-lts-next)

File: docs/source/release_notes/lts/52_0.rst [hunk 1]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+All the previously supported ways to install Avocado are still valid and
+few new ones were added, but beware that Avocado was split into several
+optional plugins so you might want to adjust your scripts/workflows.
+
+* Multiplexer (the YAML parser which used to generate variants) was
+  turned into an optional plugin ``yaml_to_mux`` also known as
+  ``avocado_framework_plugin_varianter_yaml_to_mux``. Without it Avocado
+  does not require PyYAML, but you need it to support the parsing of
+  YAML files to variants (unless you use a different plugin with similar
+  functionality, which is now also possible).
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 2]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* The HTML result plugin is now also an optional plugin so one has to
+  install it separately.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 3]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* The remote execution features (``--remote-hostname``, ``--vm-domain``,
+  ``--docker``) were also turned into optional plugins so if you need
+  those you need to install them separately.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 4]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* Support for virtual environment (``venv``) was greatly improved and we
+  do encourage people who want to use ``pip`` to do that via this method.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.  I'd add that it's now a supported installation method... and it's there on the next section.

File: docs/source/release_notes/lts/52_0.rst [hunk 5]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+
+As for the supported ways:
+
+* Fedora/RHEL can use our custom repositories, either LTS-only or all
+  releases. Note that latest versions (non-lts) are also available
+  directly in Fedora and also in EPEL.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 6]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* OpenSUSE - Ships the 36 LTS versions, hopefully they'll start
+  shipping the 52 ones as well (but we are not in charge of that
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: I'd move this to "unsupported installation methods"  section.  We're not in control, so we can not support them.

File: docs/source/release_notes/lts/52_0.rst [hunk 7]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+  process)
+* Debian - The ``contrib/packages/debian`` script is still available,
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: This one too ("unsupported installation methods").

File: docs/source/release_notes/lts/52_0.rst [hunk 8]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+  although un-maintained for a long time
+* PyPI/pip - Avocado as well as all optional plugins are available in
+  PyPI and can be installed via ``pip install avocado-framework\*``,
+  or selectively one by one.
+* From source - Makefile target ``install`` is still available but it
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: This is fine, although IMO we can also consider `make develop` a supported "installation/usage" mechanism, with its caveats stated.  What do you think?

If you agree, then a note that `make develop` will activate all the optional plugins shipped with Avocado (and won't do anything to the system) may be a good idea.

File: docs/source/release_notes/lts/52_0.rst [hunk 9]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+
+Most workflows should work the same, although there are few little
+changes and a few obsoleted constructs which are still valid, but you
+should start using the new ones.
+
+The hard changes which does not provide backward compatibility:
+
+* Human result was tweaked a bit:
+    * The ``TESTS`` entry (displaying number of tests) was removed as
+      one can easily get this information from ``RESULTS``.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 10]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+    * Instead of tests time (sum of test times) you get job time
+      (duration of the job execution) in the human result
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 11]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* Json results also contain some changes:
+    * They are pretty-printed
+    * As ``cancel`` status was introduced, json result contain an entry
+      of number of canceled tests (``cancel``)
+    * url was renamed to id (url entry is to be removed in ``53.0`` so
+      this is actually a soft change with a backward compatibility support)
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 12]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* The ``avocado multilex|variants`` does not expect multiplex YAML
+  files as positional arguments, one has to use ``-m|--mux-yaml``
+  followed by one or more paths.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK, I had in mind that the syntax was still multiple `-m FILE`.  Nice.

File: docs/source/release_notes/lts/52_0.rst [hunk 13]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* Test variants are not serialized numbers anymore in the default
+  ``yaml_to_mux`` (``multiplexer``), but ordered list of leaf-node names
+  of the variant followed by hash of the variant content
+  (paths+environment). Therefor instead of ``my_test:1`` you can get
+  something like ``my_test:arm64-virtio_scsi-RHEL7-4a3c``.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 14]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* ``results.tap`` is now generated by default in job results along the
+  ``results.json`` and ``results.xml`` (unless disabled)
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 15]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* The ``avocado run --replay`` and ``avocado diff`` are unable to parse
+  results generated by ``36.4`` to this date. We should be able to
+  introduce such feature with not insignificant effort, but no one was
+  interested yet.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 16]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+  suite the possible multiple varianter plugins:
+
+    * Instead of picking between ``tree`` representation of list of
+      variants one can use ``--summary``, resp ``--variants`` followed
+      by verbosity, which supersedes ``-c|contents``, ``-t|--tree``,
+      ``-i|--inherit``
+    * Instead of ``--filter-only|--filter-out`` the
+      ``--mux-filter-only|--mux-filter-out`` are available
+    * The ``--mux-path`` is now also available in
+      ``avocado multiplex|variants``
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK to this whole bunch of items.

File: docs/source/release_notes/lts/52_0.rst [hunk 17]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+
+Test API
+^^^^^^^^
+
+Main features stayed the same, there are few new ones so do check our
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: Suggestion from a non-native speaker: "there are **a** few new ones..."

File: docs/source/release_notes/lts/52_0.rst [hunk 18]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+documentation for details. Anyway while porting tests you should pay
+attention to following changes:
+
+* If you were overriding ``avocado.Test`` attributes (eg. ``name``,
+  ``params``, ``runner_queue``, ...) you'll get an
+  ``AttributeError: can't set attribute`` error as most of them were
+  turned into properties to avoid accidental override of the important
+  attributes.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 19]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* The ``tearDown`` method is now executed almost always (always when the
+  ``setUp`` is entered), including when the test is interrupted while
+  running ``setUp``. This might require some changes to your ``setUp``
+  and ``tearDown`` methods but generally it should make them simpler.
+  (See :ref:`Setup and cleanup methods` and following chapters for details)
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 20]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* Test exceptions are publicly available directly in ``avocado``
+  (``TestError``, ``TestFail``, ``TestCancel``) and when raised inside
+  test they behave the same way as ``self.error``, ``self.fail`` or
+  ``self.cancel``. (See :mod:`avocado`)
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 21]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* New status is available called ``CANCEL``. It means the test (or even
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: "A new status named ``CANCEL`` is now available."

File: docs/source/release_notes/lts/52_0.rst [hunk 22]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+  similar to ``SKIP`` in other frameworks, but the ``SKIP`` result is
+  reserved for tests that were not executed (nor the ``setUp`` was
+  entered). The ``CANCEL`` status can be signaled by ``self.cancel`` or
+  by raising ``avocado.TestCancel`` exception and the ``SKIP`` should be
+  set only by ``avocado.skip``, ``avocado.skipIf`` or
+  ``avocado.skipUnless`` decorators. The ``self.skip`` method is still
+  supported but will be removed after in ``53.0`` so you should replace
+  it by ``self.cancel`` which has similar meaning but it additionally
+  executes the ``tearDown``. (See :ref:`Test statuses`
+* The ``tag`` argument of ``avocado.Test`` was removed as it is part of
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 23]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+  ``name``, which can only be ``avocado.core.test.TestName`` instance.
+  (See :meth:`avocado.core.test.Test`)
+* The ``self.job.logdir`` which used to be abused to share state/data
+  between tests inside one job can now be dropped towards the
+  ``self.teststmpdir``, which is a shared temporary directory which
+  sustains throughout job execution and even between job executions if
+  set via ``AVOCADO_TESTS_COMMON_TMPDIR`` environmental value. (See
+  :meth:`avocado.core.test.Test.teststmpdir`)
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 24]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* Those who write inherited test classes will be pleasantly surprised as
+  it is now possible to mark a class as avocado test including all
+  ``test*`` methods coming from all parent classes (similarly to how
+  dynamic discovery works inside python unittest, see
+  :ref:`docstring-directive-recursive` for details)
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK.

File: docs/source/release_notes/lts/52_0.rst [hunk 25]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* The ``self.text_output`` is not published after the test execution. If
+  you were using it simply open the ``self.logfile`` and read the content
+  yourself.
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: I honestly had no memories about `self.text_output`.

File: docs/source/release_notes/lts/52_0.rst [hunk 26]
```diff
@@ -163,3 +163,176 @@ When compared to the last LTS, we had:
...
+* ``avocado.utils.build.make`` calls as it now reports only
+  ``exit_status``. To get the full result object you need to execute
+  ``avocado.utils.build.run_make``.
+* ``avocado.utils.distro`` reports ``Red Hat Enterprise Linux``/``rhel``
+  instead of ``Red Hat``/``redhat``.
+* ``avocado.process`` where the check for availability of ``sudo`` was
+  improved, which might actually start executing some code which used
+  to fail in 36.4.
+
+Also check out the :mod:`avocado.utils` for complete list of available
```
Comment (clebergnu) [inline docs/source/release_notes/lts/52_0.rst]: OK!

File: docs/source/release_notes/lts/52_0.rst (hunk 6 above)
Comment (ldoktor) [inline docs/source/release_notes/lts/52_0.rst]: I'm not listing supported methods, I'm letting people aware of what is out there and that they should check that out before going to pip (as distro-methods are always preferred)

File: docs/source/release_notes/lts/52_0.rst (hunk 7 above)
Comment (ldoktor) [inline docs/source/release_notes/lts/52_0.rst]: This is actually important as it probably worked at 36 so people might have been using it, but now they should really pay attention to adjust their scripts. This is not list of supported variants, but list of what changed...

File: docs/source/release_notes/lts/52_0.rst (hunk 8 above)
Comment (ldoktor) [inline docs/source/release_notes/lts/52_0.rst]: I don't consider this for public use, it's meant for developers... so I'd prefer keeping it this way, it's not "how to install" section, but what changed concerning installation.

File: docs/source/release_notes/lts/52_0.rst (hunk 12 above)
Comment (ldoktor) [inline docs/source/release_notes/lts/52_0.rst]: nop, on the other hand that's why you need the `--` to separate files and positional arguments ;-) (eg. `avocado run -m foo passtest.py` fails, you have to use `avocado run -m foo -- passtest.py`).

File: docs/source/release_notes/lts/52_0.rst (hunk 17 above)
Comment (ldoktor) [inline docs/source/release_notes/lts/52_0.rst]: `a few` means few but significant amount, while `few` means insignificant amount. (at least that's what I had been told). Here I wanted to stress out it's really just minimal number of changed things.

File: docs/source/release_notes/lts/52_0.rst (hunk 21 above)
Comment (ldoktor) [inline docs/source/release_notes/lts/52_0.rst]: OK, will fix in v2

**Pull request closed without merging**